sa/storage-authority.go

```diff
@@ -56,28 +62,28 @@ func (ssa *SQLStorageAuthority) InitTables() (err error) {
 	}
 
 	// Create registrations table
-	_, err = tx.Exec("CREATE TABLE registrations (id TEXT, thumbprint TEXT, value TEXT);")
+	_, err = tx.Exec("CREATE TABLE IF NOT EXISTS registrations (id TEXT, thumbprint TEXT, value TEXT);")
 	if err != nil {
 		tx.Rollback()
 		return
 	}
 
 	// Create pending authorizations table
-	_, err = tx.Exec("CREATE TABLE pending_authz (id TEXT, value BLOB);")
+	_, err = tx.Exec("CREATE TABLE IF NOT EXISTS pending_authz (id TEXT, value BLOB);")
 	if err != nil {
 		tx.Rollback()
 		return
 	}
 
 	// Create finalized authorizations table
-	_, err = tx.Exec("CREATE TABLE authz (sequence INTEGER, id TEXT, digest TEXT, value BLOB);")
+	_, err = tx.Exec("CREATE TABLE IF NOT EXISTS authz (sequence INTEGER, id TEXT, digest TEXT, value BLOB);")
 	if err != nil {
 		tx.Rollback()
 		return
 	}
 
 	// Create certificates table
-	_, err = tx.Exec("CREATE TABLE certificates (serial string, digest TEXT, value BLOB);")
+	_, err = tx.Exec("CREATE TABLE IF NOT EXISTS certificates (serial INTEGER, digest TEXT, value BLOB);")
```

Comment:
We're storing this as a hex-encoded string for a couple of reasons. Mainly that it's easier to eyeball, since there's some internal structure to the serial: The first 8 bits are an intermediate identifier, the next 56 bits are a sequential counter, and the final 64 bits are random. It's useful to be able to eyeball these in the DB. It's also nice to be able to easily query for serials with a given prefix, in order to enumerate all certs by the counter part. So I'd like this to stay as a string.